### PR TITLE
feat: ems can place player in bed

### PR DESCRIPTION
**Pull request by solareon**
## Description

Added feature to allow ems player to put another player in a bed. Also added feature to automatically set duty status to false when in laststand to prevent ems from being stuck in a bed

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My code fits the style guidelines.
- [x] My PR fits the contribution guidelines.

### CODE REVIEW

File: server/main.lua [hunk 1]
```diff
@@ -109,6 +109,9 @@ end)
 RegisterNetEvent('qbx_medical:server:onPlayerLaststand', function(text)
 	if GetInvokingResource() then return end
 	local src = source
+    local player = exports.qbx_core:GetPlayer(src)
+	player.Functions.SetJobDuty(false)
+    TriggerClientEvent('QBCore:Client:SetDuty', src, player.PlayerData.job.onduty)
```
Comment (Manason) [inline server/main.lua]: This should be triggered by core. Not here.

File: client/hospital.lua [hunk 2]
```diff
@@ -157,6 +161,24 @@ if config.useTarget then
...
+                                if QBX.PlayerData.job.name == 'ambulance' then
+                                    return true
+                                end
+                                return false
+                            end,
+                            onSelect = function()
+                                local player = GetClosestPlayer()
+                                if player then
+                                    local playerId = GetPlayerServerId(player)
+                                    TriggerServerEvent('hospital:server:putPlayerInBed', playerId, hospitalName, i)
```
Comment (Manason) [inline client/hospital.lua:176]: It doesn't look like this server event does anything but call this client event to put a player in bed. Why not just call the function putPlayerInBed here directly?

**Review by Manason — CHANGES_REQUESTED** [review]

File: server/main.lua (hunk 1 above)
Comment (solareon) [reply server/main.lua]: removed

File: server/main.lua [hunk 3]
```diff
@@ -150,6 +150,15 @@ RegisterNetEvent('hospital:server:RevivePlayer', function(playerId)
 	TriggerClientEvent('hospital:client:Revive', patient.PlayerData.source)
 end)
 
+---@param playerId number
+RegisterNetEvent('hospital:server:putPlayerInBed', function(playerId, hospitalName, bedIndex)
+	if GetInvokingResource() then return end
+	local patient = exports.qbx_core:GetPlayer(playerId)
+
+	if not patient then return end
+	TriggerClientEvent('qbx_ambulancejob:client:putPlayerInBed', patient.PlayerData.source, hospitalName, bedIndex)
```
Comment (Manason) [inline server/main.lua]: No need to get the player object just for the source
```suggestion
	TriggerClientEvent('qbx_ambulancejob:client:putPlayerInBed', playerId, hospitalName, bedIndex)
```

File: client/hospital.lua [hunk 4]
```diff
@@ -157,6 +161,24 @@ if config.useTarget then
                             icon = "fas fa-clipboard",
                             label = Lang:t('text.bed'),
                             distance = 1.5,
+                        },
+                        {
+                            canInteract = function()
+                                if QBX.PlayerData.job.name == 'ambulance' then
+                                    return true
+                                end
+                                return false
```
Comment (Manason) [inline client/hospital.lua]:
```suggestion
                                return QBX.PlayerData.job.name == 'ambulance'
```

**Review by Manason — APPROVED** [review]

**Pull request merged**